hooks/useEscrowedEXA.ts

```diff
@@ -0,0 +1,116 @@
+import { useCallback, useEffect, useState } from 'react';
+import { Address, zeroAddress } from 'viem';
+import {
+  escrowedExaABI,
+  useEscrowedExaBalanceOf,
+  useEscrowedExaReserveRatio,
+  useEscrowedExaReserves,
+  useEscrowedExaVestingPeriod,
+} from 'types/abi';
+import useContract from './useContract';
+import { useEXA } from './useEXA';
+import useGraphClient from './useGraphClient';
+import { useWeb3 } from './useWeb3';
+import { getStreams } from 'queries/getStreams';
+
+export const useEscrowedEXA = () => {
+  return useContract('EscrowedEXA', escrowedExaABI);
+};
+
+type Stream = {
+  id: string;
+  tokenId: string;
+  recipient: Address;
+  startTime: string;
+  endTime: string;
+  depositAmount: string;
+  withdrawnAmount: string;
+  canceled: boolean;
+  cancelable: boolean;
+};
+
+export function useUpdateStreams() {
+  const EXA = useEXA();
+  const esEXA = useEscrowedEXA();
+  const { walletAddress } = useWeb3();
+  const request = useGraphClient();
+
+  const [activeStreams, setActiveStreams] = useState<Stream[]>([]);
+  const [loading, setLoading] = useState<boolean>(true);
+
+  const fetchStreams = useCallback(async () => {
+    if (EXA && esEXA) {
+      setLoading(true);
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      const data = await request<any>(
```

Comment:
why are you typing as any? you're declaring the type below when filtering `data.streams`.
```
      type StreamsResponse = {
        streams: { startTime: string; duration: string; endTime: string; intactAmount: string }[];
      }[];

      const data = await request<StreamsResponse>(...
```